### PR TITLE
Migrate to gha

### PR DESCRIPTION
**Pull request by ayeminag**
### Tickets
- [#6960](https://github.com/signalwire/cloud-product/issues/6960)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    steps:
+      - name: install deps
+        run: |
+          apt-get update && apt-get install -yq build-essential autotools-dev lsb-release pkg-config automake autoconf libtool-bin clang-tools-7
+          apt-get install -yq cmake uuid-dev libssl-dev
+      - name: Checkout
+        uses: actions/checkout@v3
+      - run: pwd
+      - id: scan_build
+        run: /__w/libks/libks/scan_build.sh
```
Comment (stephane-alnet) [inline .github/workflows/ci.yml:22]: this feels horribly wrong, shouldn't we be able to do `./scan_build.sh`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Tar logs
+        id: tar
+        if: failure()
+        env:
+          COMPILATION_FAILED: ${{ steps.scan_build.outputs.COMPILATION_FAILED }}
+          BUGS_FOUND: ${{ steps.scan_build.outputs.BUGS_FOUND }}
+        run: |
+          cd /__w/libks/libks
```
Comment (stephane-alnet) [inline .github/workflows/ci.yml:30]: is this needed?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          ls -l
+          if [ "true" -eq $COMPILATION_FAILED ]; then
+            tar czvf /__w/libks/libks/scan-build-result.tar.gz ./scan-build-result.txt;
```
Comment (stephane-alnet) [inline .github/workflows/ci.yml:33]:
```suggestion
            tar czvf scan-build-result.tar.gz ./scan-build-result.txt;
```

File: scan_build.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+cd /__w/libks/libks
```
Comment (stephane-alnet) [inline scan_build.sh:2]:
```suggestion
```

File: run_tests.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+cd /__w/libks/libks
```
Comment (stephane-alnet) [inline run_tests.sh:2]:
```suggestion
```

**Review by stephane-alnet — APPROVED** [review]
having to `cd` to what should be the default directory sounds wrong? (I didn't comment on all instances)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ayeminag) [reply .github/workflows/ci.yml]: fixed

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ayeminag) [reply .github/workflows/ci.yml]: removed

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ayeminag) [reply .github/workflows/ci.yml]: removed

File: scan_build.sh (hunk 4 above)
Comment (ayeminag) [reply scan_build.sh]: removed

File: run_tests.sh (hunk 5 above)
Comment (ayeminag) [reply run_tests.sh]: removed

**Pull request merged**